### PR TITLE
Test tools

### PR DESCRIPTION
**Pull request by mikurys**

### CODE REVIEW

File: src/tunserver.cpp [hunk 1]
```diff
@@ -368,27 +368,108 @@ c_tunnel_use::c_tunnel_use(const antinet_crypto::c_multikeys_PAIR & ID_self,
...
+	try {
+		c_galaxysrv_peers::t_peering_reference_parse parse = c_galaxysrv_peers::parse_peer_reference(simple); // partially parsed
+		_check_input(parse.first.size() > 0);
+		_check_input(parse.second.size() > 0);
+		std::string cable = parse.second[0];
+		// udp:194.28.50.88:19042
+		auto begin = cable.find(':');
+		_check_input(begin != std::string::npos);
+		std::string ipv4_with_port = cable.substr(begin+1);
+        auto end = ipv4_with_port.find(':');
```
Comment (robertoleksy) [inline src/tunserver.cpp]: tabs

File: src/tunserver.cpp [hunk 2]
```diff
@@ -368,27 +368,108 @@ c_tunnel_use::c_tunnel_use(const antinet_crypto::c_multikeys_PAIR & ID_self,
...
+		_throw_error( std::invalid_argument(mo_file_reader::gettext("L_bad_peer_format")) );
+	}
+}
+
+void c_tunserver::delete_all_peers(bool is_banned)
+{
+	_mark("delete all peers (delete only!) ");
+	{
+		LockGuard<Mutex> lg(m_peer_mutex);
+		LockGuard<Mutex> lg_block_list(m_peer_black_list_mutex);
```
Comment (robertoleksy) [inline src/tunserver.cpp]: typo? block

File: src/tunserver.cpp [hunk 3]
```diff
@@ -926,6 +1045,120 @@ string c_tunserver::rpc_peer_list(const string &input_json) {
...
+string c_tunserver::rpc_delete_peer(const string &input_json)
+{
+	auto input = nlohmann::json::parse(input_json);
+	auto peer = input["peer"].get<std::string>();
+	nlohmann::json ret;
+	ret["cmd"] = "delete_peer";
+	try{
+		delete_peer_simplestring(peer, false);
+		ret["msg"] = "peer deleted";
+	} catch(const std::invalid_argument &ex) {
```
Comment (robertoleksy) [inline src/tunserver.cpp]: unused ex

File: src/tunserver.cpp [hunk 4]
```diff
@@ -926,6 +1045,120 @@ string c_tunserver::rpc_peer_list(const string &input_json) {
...
+string c_tunserver::rpc_ban_peer(const string &input_json)
+{
+	auto input = nlohmann::json::parse(input_json);
+	auto peer = input["peer"].get<std::string>();
+	nlohmann::json ret;
+	ret["cmd"] = "ban_peer";
+	try{
+		delete_peer_simplestring(peer, true);
+		ret["msg"] = "peer banned";
+	} catch(const std::invalid_argument &ex) {
```
Comment (robertoleksy) [inline src/tunserver.cpp]: unused ex

File: src/tunserver.cpp (hunk 1 above)
Comment (mikurys) [reply src/tunserver.cpp]: 👍

File: src/tunserver.cpp (hunk 2 above)
Comment (mikurys) [reply src/tunserver.cpp]: 👍

File: src/tunserver.cpp (hunk 4 above)
Comment (mikurys) [reply src/tunserver.cpp]: 🆗

**Review by robertoleksy — APPROVED** [review]

**Pull request merged**